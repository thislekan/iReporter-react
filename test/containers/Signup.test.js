import '@babel/polyfill';
import React from 'react';
import setup from '../mock/setup';
import ConnectedSignup from '../../src/components/Signup.jsx';

const mockFn = jest.fn();
const history = { push: mockFn };

describe('<UserForm />', () => {
  it('Should render all elements', () => {
    const wrapper = setup(<ConnectedSignup history={history} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should respond to input change and alter state (email) of component', () => {
    const wrapper = setup(<ConnectedSignup history={history} />);
    const signupWrapper = wrapper.find('Signup');
    signupWrapper.find('#email').simulate('change', { target: { name: 'email', value: 'wrestle@wwe.com' } });
    expect(signupWrapper.state('email')).toBe('wrestle@wwe.com');
    expect(wrapper.find('#email').props().value).toBe('wrestle@wwe.com');
  });
  it('should submit form', () => {
    const wrapper = setup(<ConnectedSignup history={history} />);
    const signupWrapper = wrapper.find('Signup');
    signupWrapper.find('form').simulate('submit', { preventDefault: mockFn });
    expect(mockFn).toHaveBeenCalled();
  });
});
