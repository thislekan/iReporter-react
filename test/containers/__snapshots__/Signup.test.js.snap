// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> Should render all elements 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Signup)
        history={
          Object {
            "push": [MockFunction],
          }
        }
      >
        <Signup
          authStatus="failure"
          authenticateUser={[Function]}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          isAdmin={false}
          isLoading={false}
          message=""
        >
          <div>
            <div>
              <div>
                <div>
                  <img
                    alt=""
                    src={Object {}}
                  />
                </div>
                <div
                  id="form"
                >
                  <div>
                    <div>
                      <div>
                        <Link
                          to="/signup"
                        >
                          <a
                            href="/signup"
                            onClick={[Function]}
                          >
                            Sign Up
                          </a>
                        </Link>
                      </div>
                      <div>
                        <Link
                          to="/login"
                        >
                          <a
                            href="/login"
                            onClick={[Function]}
                          >
                            Log In
                          </a>
                        </Link>
                      </div>
                    </div>
                    <form
                      onSubmit={[Function]}
                    >
                      <div>
                        <label
                          htmlFor="name"
                        >
                          LAST NAME
                        </label>
                        <input
                          id="last-name"
                          name="lastName"
                          onBlur={[Function]}
                          placeholder="Enter your last name"
                          type="text"
                        />
                      </div>
                      <div>
                        <label
                          htmlFor="name"
                        >
                          FIRST NAME
                        </label>
                        <input
                          id="first-name"
                          name="firstName"
                          onBlur={[Function]}
                          placeholder="Enter your full name"
                          type="text"
                        />
                      </div>
                      <div>
                        <label
                          htmlFor="email"
                        >
                          EMAIL
                        </label>
                        <input
                          id="email"
                          name="email"
                          onBlur={[Function]}
                          placeholder="Enter your email address"
                          type="email"
                        />
                      </div>
                      <div>
                        <label
                          htmlFor="password"
                        >
                          PASSWORD
                        </label>
                        <input
                          id="password"
                          name="password"
                          onBlur={[Function]}
                          placeholder="******"
                          type="password"
                        />
                      </div>
                      <div>
                        <label
                          htmlFor="confirm password"
                        >
                          CONFIRM PASSWORD
                        </label>
                        <input
                          id="confirm-password"
                          name="confirmPassword"
                          onBlur={[Function]}
                          placeholder="******"
                          type="password"
                        />
                      </div>
                      <div
                        className="btn-div"
                      >
                        <button
                          id="signup-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Signup>
      </Connect(Signup)>
    </Router>
  </BrowserRouter>
</Provider>
`;
